src/Castle.Facilities.AspNetCore/WindsorRegistrationExtensions.cs

```diff
@@ -93,35 +93,56 @@ public static ComponentRegistration AsMiddleware(this ComponentRegistration regi
 
 		private static void AddApplicationComponentsToWindsor(IWindsorContainer container, WindsorRegistrationOptions options)
 		{
-			if (!options.ControllerRegistrations.Any())
+			// Applying default registrations, Lifestyle here is Scoped
+
+			if (!options.ControllerAssemblyRegistrations.Any() && !options.ControllerComponentRegistrations.Any())
 			{
-				container.Register(Classes.FromAssemblyInThisApplication(options.EntryAssembly).BasedOn<Controller>().LifestyleScoped());
+				container.Register(Classes.FromAssemblyInThisApplication(options.EntryAssembly).BasedOn<ControllerBase>().LifestyleScoped());
 			}
 
-			foreach (var controllerRegistration in options.ControllerRegistrations)
+			if (!options.TagHelperAssemblyRegistrations.Any() && !options.TagHelperComponentRegistrations.Any())
 			{
-				container.Register(Classes.FromAssemblyInThisApplication(controllerRegistration.Item1).BasedOn<Controller>().Configure(x => x.LifeStyle.Is(controllerRegistration.Item2)));
+				container.Register(Classes.FromAssemblyInThisApplication(options.EntryAssembly).BasedOn<ViewComponent>().LifestyleScoped());
 			}
 
-			if (!options.TagHelperRegistrations.Any())
+			if (!options.ViewComponentAssemblyRegistrations.Any() && !options.ViewComponentComponentRegistrations.Any())
 			{
-				container.Register(Classes.FromAssemblyInThisApplication(options.EntryAssembly).BasedOn<ViewComponent>().LifestyleScoped());
+				container.Register(Classes.FromAssemblyInThisApplication(options.EntryAssembly).BasedOn<ITagHelper>().LifestyleScoped());
 			}
 
-			foreach (var controllerRegistration in options.TagHelperRegistrations)
+			// Assembly registrations
+
+			foreach (var controllerRegistration in options.ControllerAssemblyRegistrations)
 			{
-				container.Register(Classes.FromAssemblyInThisApplication(controllerRegistration.Item1).BasedOn<ViewComponent>().Configure(x => x.LifeStyle.Is(controllerRegistration.Item2)));
+				container.Register(Classes.FromAssemblyInThisApplication(controllerRegistration.Item1).BasedOn<Controller>().Configure(x => x.LifeStyle.Is(controllerRegistration.Item2)));
```

Comment:
Is this one still meant to be `Controller`?

Reply:
Whoops, will fix.